### DISCUSSION TITLE
RTC for Sizif-512

### DISCUSSION THREAD

**Discussion by LuckStilon** [UzixLS/zx-sizif-512#63] [General]
I once had an idea to make a cheap RTC clock for Sizif. First, by connecting a module, e.g. DS1307 (I2C), to the AY outputs via the J13 connector. However, you would need to write an RTC.SYS driver for this that would be able to communicate through the AY outputs using the I2C protocol. It's difficult, because driver code has to fit into 256 bytes. 

I once made a similar device and wrote an RTC.SYS driver for it. But that solution is too expensive:
https://mumio.dev/?p=135

**But then I came up with another idea to do it backwards!**:
Add an I2C bus to Sizif, accessible through ports 103B and 113B. That's will be exactly compatible with ZX Spectrum Next. Then, you would **not need to write your own RTC.SYS** driver to support RTC, because you could immediately use the same one used in Next. This way, you could add other I2C devices that already have drivers written to this bus for Next.

A bus can be added as expansion card to the southern ZXBUS (J18) or - alternatively - built into the PCB design.

@UzixLS What do You think about it?

**Comment by UzixLS** [comment]
Nice idea :) If you plan to implement ports within CPLD, you can use testpoints around to bring out I2C bus.

**Reply by LuckStilon** [reply]
O! Nice idea, to use CPLD. I see free pins 94-96. Are they intended for something, or can they be used for I2C?

**Reply by UzixLS** [reply]
They aren't in use currently
